src/database/memory.rs

```diff
@@ -486,15 +486,23 @@ impl ConfigurableDatabase for MemoryDatabase {
 /// don't have `test` set.
 macro_rules! populate_test_db {
     ($db:expr, $tx_meta:expr, $current_height:expr$(,)?) => {{
+        $crate::populate_test_db!($db, $tx_meta, $current_height, (@coinbase false))
+    }};
+    ($db:expr, $tx_meta:expr, $current_height:expr, (@coinbase $is_coinbase:expr)$(,)?) => {{
         use std::str::FromStr;
         use $crate::database::BatchOperations;
         let mut db = $db;
         let tx_meta = $tx_meta;
         let current_height: Option<u32> = $current_height;
+        let input = if $is_coinbase {
+            vec![$crate::bitcoin::TxIn::default()]
+        } else {
+            vec![]
```

Comment:
Not that its highly important but here the input remains always empty even if its there in `tx_meta`.. This doesn't fail any db tests because we never deal with inputs, but might make sense to get the inputs too from `tx_meta` for sake of completeness.. I have done it in this PR here https://github.com/bitcoindevkit/bdk/pull/624/files

Something like this 
```
        let input = if is_coinbase {
            // `TxIn::default()` creates a coinbase input, by definition.
            vec![TxIn::default()]
        } else {
            tx_meta
                .input
                .iter()
                .map(|test_input| {
                    let mut txin = TxIn::default();
                    txin.previous_output = OutPoint {
                        txid: test_input.txid,
                        vout: test_input.vout,
                    };

                    if let Some(seq) = test_input.sequence {
                        txin.sequence = seq;
                    }
                    txin
                })
                .collect()
        };
  ```
Not blocking on this PR. Can be done in a separate one later..